go/lib.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+ */
+
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+typedef enum RequestErrorType {
+  Unspecified = 0,
+  ExecAbort = 1,
+  Timeout = 2,
+  Disconnect = 3,
+} RequestErrorType;
+
+/**
+ * The connection response.
+ *
+ * It contains either a connection or an error. It is represented as a struct instead of an enum for ease of use in the wrapper language.
```

Comment:
Did you mean union?
```suggestion
 * It contains either a connection or an error. It is represented as a struct instead of a union for ease of use in the wrapper language.
```